src/main/java/com/yelp/nrtsearch/server/luceneserver/field/TermQueryableIndexableFieldDef.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright 2020 Yelp Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.yelp.nrtsearch.server.luceneserver.field;
+
+import com.yelp.nrtsearch.server.grpc.Field;
+import com.yelp.nrtsearch.server.grpc.TermInSetQuery;
+import com.yelp.nrtsearch.server.grpc.TermQuery;
+import com.yelp.nrtsearch.server.luceneserver.field.properties.TermQueryable;
+import org.apache.lucene.search.Query;
+
+public abstract class TermQueryableIndexableFieldDef extends IndexableFieldDef
+    implements TermQueryable {
+  /**
+   * Field constructor. Performs generalized building of field definition by calling a number of
+   * protected methods, mainly: {@link #validateRequest(Field)}, {@link #parseDocValuesType(Field)},
+   * {@link #parseFacetValueType(Field)}, and {@link #setSearchProperties(FieldType, Field)}.
+   * Concrete field types should override these methods as needed.
+   *
+   * @param name name of field
+   * @param requestField field definition from grpc request
+   */
+  protected TermQueryableIndexableFieldDef(String name, Field requestField) {
+    super(name, requestField);
+  }
+
+  // Performs type validation. Do not @Override in subclasses.
+  @Override
+  public Query getTermQuery(TermQuery termQuery) {
+    switch (termQuery.getTermTypesCase()) {
+      case BOOLEANVALUE:
+        return getTermQueryFromBooleanValue(termQuery);
+      case DOUBLEVALUE:
+        return getTermQueryFromDoubleValue(termQuery);
+      case FLOATVALUE:
+        return getTermQueryFromFloatValue(termQuery);
+      case INTVALUE:
+        return getTermQueryFromIntValue(termQuery);
+      case LONGVALUE:
+        return getTermQueryFromLongValue(termQuery);
+      case TEXTVALUE:
+        return getTermQueryFromTextValue(termQuery);
+      default:
+        throw new IllegalArgumentException(
+            String.format(
+                "%s field does not support term type: %s",
+                termQuery.getField(), termQuery.getTermTypesCase()));
+    }
+  }
+
+  Query getTermQueryFromBooleanValue(TermQuery termQuery) {
```

Comment:
Since you already know which value of the query message is set, you could probably just pass that as the function parameter, i.e. the bool/int/float/double/text value. This would also allow the name to be shortened, as the parameter would define the type.

Reply:
Changed to pass in the expected value type.

Can't shorten function names to `getTermInSetQuery(<type>)` because List<type> can't be overloaded (https://stackoverflow.com/questions/2241514/overloading-java-function-with-list-parameter). Kept `getTermQueryFrom<Type>Value(<type>)` for consistency with this.